aws_lambda_powertools/utilities/validation/base.py

```diff
@@ -29,7 +39,9 @@ def validate_data_against_schema(data: Union[Dict, str], schema: Dict, formats:
     """
     try:
         formats = formats or {}
-        fastjsonschema.validate(definition=schema, data=data, formats=formats)
+        handlers = handlers or {}
+        provider_options = provider_options or {}
```

Comment:
So... it looks like this last line
```
provider_options = provider_options or {}
```
is what bumps the xenon complexity from A to B.

Any thoughts on how to resolve this? I haven't worked with this sort of constraint -- this additional line seems pretty reasonable to me.

Reply:
Looks like the `complexity-baseline` task in the Makefile manually excludes `aws_lambda_powertools/shared/json_encoder.py`? Seems a bit hacky to skip the check that way, though the complexity checker also seem a bit aggressive. I can add this file to the skipped files if that's how we want to resolve this failed check.

Reply:
Hey @danieljnchen, this is because you added this new field (`provider_options = provider_options or {}`) with a conditional and Xenon/Radon is complaining about cyclomatic complexity.
I agree that the check is a bit aggressive in this case and we can skip it manually. Let me review the PR and I'll leave a message with the decision, ok?